### ISSUE TITLE
Extend the `no-identical-title` rule to support `suite` test block pattern

### ISSUE THREAD

**Issue by ta9yu** [veritem/eslint-plugin-vitest#385]
Hello,

Currently, the `no-identical-title` rule in eslint-plugin-vitest only covers the pattern where the test block is `describe`.

However, there is another commonly used test block pattern that is not covered by this rule. Specifically, the pattern where the test block is `suite` is not recognized by the rule.

It would be a great improvement to the utility of this plugin if the `no-identical-title` rule could be extended to support the `suite` test block pattern.

Thank you for considering this suggestion.

Best regards.

**Comment by veritem** [comment]
@ta9yu would you mind sharing an example of a `suite` test block you're talking about? I'm not getting exactly what you mean but `suite` test block pattern.

**Comment by yshrsmz** [comment]
@veritem 

There is a `suite()` api in vitest.

https://github.com/vitest-dev/vitest/blob/6d1b145118d42b14bc0739820c4ce9e17fbd45dd/packages/runner/src/suite.ts#L12

and `describe()` is an alias for that.

```ts
describe('parent', () => {
  describe('child 1', () => {
    test('grand child 1', () => {})
  })

  describe('child 2', () => {
    test('grand child 1', () => {})
  })
})
```

While the above code does not warn about `no-identical-title`, below code does warn about it.

```ts
suite('parent', () => {
  suite('child 1', () => {
    test('grand child 1', () => {})
  })

  suite('child 2', () => {
    test('grand child 1', () => {})
  })
})
```

**Comment by veritem** [comment]
@yshrsmz Thanks, I didn't know that. That is my bad!

**Comment by ta9yu** [comment]
@veritem 

Thank you for your quick response and action.
Your efforts are greatly appreciated!
